mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp

```diff
@@ -110,6 +110,137 @@ class NvInferEngineWrapper {
+  const std::string &getTensorName() const { return mTensorName; }
+
+private:
+  /// Round up the given value m to the nearest multiple of n.
+  static inline uint64_t roundUp(uint64_t m, uint64_t n) {
+    return llvm::divideCeil(m, n) * n;
+  }
+
+  uintptr_t mOutputPtr;
+  uint64_t mOutputSize{0};
+  nvinfer1::Dims mOutputDims;
+  std::string mTensorName;
+  AllocTracker *mTracker{nullptr};
+};
+
+/// It is the responsibilty of runtime client to release the buffer.
+OutputAllocatorImpl::~OutputAllocatorImpl() {}
+
+void OutputAllocatorImpl::setTensorName(const std::string &name) {
+  mTensorName = name;
+}
+
+void OutputAllocatorImpl::setCurrentMemory(uintptr_t memory, int64_t size) {
+  mOutputPtr = memory;
+  mOutputSize = size;
+}
+
+void *OutputAllocatorImpl::reallocateOutputAsync(const char *name, void *memory,
+                                                 uint64_t size,
+                                                 uint64_t alignment,
+                                                 cudaStream_t stream) {
+  assert((!mOutputPtr || reinterpret_cast<uintptr_t>(memory) == mOutputPtr) &&
+         "Output buffer mismatch");
+  assert(name == mTensorName && "Tensor name mismatch");
+
+  size = std::max(size, static_cast<uint64_t>(1));
+
+  if (size > mOutputSize) {
+    size = roundUp(size, alignment);
+
+    if (mOutputPtr)
+      mlirtrt::runtime::safeDeallocate(*mTracker, mOutputPtr,
+                                       CudaStreamPtr(stream));
+
+    mOutputPtr = 0;
+    mOutputSize = 0;
+
+    StatusOr<PointerInfo> memory = mlirtrt::runtime::allocate(
+        *mTracker, PointerType::device, size, alignment, CudaStreamPtr(stream));
+    if (memory.isOk()) {
+      mOutputPtr = (*memory).ptr;
+      mOutputSize = memory->size;
+    }
+    return reinterpret_cast<void *>(mOutputPtr);
+  }
+  return reinterpret_cast<void *>(mOutputPtr);
+}
+
+void OutputAllocatorImpl::notifyShape(const char *name,
+                                      const nvinfer1::Dims &dims) {
+  assert(name == mTensorName && "Tensor name mismatch");
+  mOutputDims = dims;
+}
+
+/// OutputAllocator - Manages multiple OutputAllocatorImpl instances.
+class OutputAllocator {
+public:
+  explicit OutputAllocator(AllocTracker *tracker, int64_t nbResults);
+
+  // Disable copy and move operations
+  OutputAllocator(const OutputAllocator &) = delete;
+  OutputAllocator &operator=(const OutputAllocator &) = delete;
+  OutputAllocator(OutputAllocator &&) = delete;
+  OutputAllocator &operator=(OutputAllocator &&) = delete;
+
+  void registerAllocator(size_t index, const char *name, uintptr_t ptr,
+                         int64_t size, nvinfer1::IExecutionContext *context);
+  OutputAllocatorImpl *getAllocator(size_t index);
+
+private:
+  std::vector<std::unique_ptr<OutputAllocatorImpl>> mAllocators;
+};
+
+OutputAllocator::OutputAllocator(AllocTracker *tracker, int64_t nbResults) {
+  mAllocators.reserve(nbResults);
+  for (int64_t i = 0; i < nbResults; ++i) {
```

Comment:
Nit: remove braces 

https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements